### PR TITLE
Extract EC2Setup class (new)

### PR DESCRIPTION
**Pull request by mschnitzer**
and move temporary key creation to it.
EC2Preparer#prepare will also create a temporary VPC and Security group with all dependencies.
This makes sure that all dependencies are already set up when calling EC2ImageUploader.

Instead of adding this functionality to the ec2utilsutils class I extracted a new EC2Preparer class to follow the single responsibility principle. Also this class contains a lot of data (like all the temporary assets it creates) which would bloat ec2utilsutils. @rjschwei let me know what you think, I'm open for your suggestions.

_________

@rjschwei I addressed all pep8 issues and fixed a syntax error. Please have a look again. I will squash the commits before the merge. It's just for you to see what has changed.

Both files were put through `pep8`.

_________

Closes #187

### CODE REVIEW

**Comment by mschnitzer** [comment]
@rjschwei the code has been tested and it seems to work.

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 1]
```diff
@@ -437,41 +442,45 @@ try:
                                               '--type')
 
         key_pair_name = args.sshName
+        ssh_private_key_file = args.privateKey
+        if not key_pair_name and not ssh_private_key_file and not \
+                args.accountName:
```
Comment (rjschwei) [inline ec2utils/ec2uploadimg/ec2uploadimg]: Wrap this condition in (), to match our style, i.e.

if (
    not key_pair_name 
    and not ssh_private_key_file
    and not args.accountName
):

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 2]
```diff
@@ -437,41 +442,45 @@ try:
...
+            key_pair_name, ssh_private_key_file = \
+                setup.create_upload_key_pair()
```
Comment (rjschwei) [inline ec2utils/ec2uploadimg/ec2uploadimg]: Should fit on the line above

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 3]
```diff
@@ -549,3 +564,6 @@ except EC2UploadImgException as e:
 except Exception as e:
     print(format(e), file=sys.stderr)
     sys.exit(1)
+finally:
+    if setup:
+        setup.clean_up()
```
Comment (rjschwei) [inline ec2utils/ec2uploadimg/ec2uploadimg]: clean up should be unconditional, we create the object unconditionally. Also cleanup needs to be in the region loop or we will leave traces behind in each region but the last. clean_up call also needs to be in the exception handlers.

**Review by rjschwei — CHANGES_REQUESTED** [review]

File: ec2utils/ec2uploadimg/ec2uploadimg (hunk 2 above)
Comment (mschnitzer) [reply ec2utils/ec2uploadimg/ec2uploadimg]: Nope as it's `80` characters long. The default line length limit is `79` characters.

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 4]
```diff
@@ -366,207 +366,208 @@ try:
...
-                print('Could not determine helper AMI-ID', file=sys.stderr)
-                sys.exit(1)
-        bootkernel = args.akiID
-        if args.virtType == 'hvm':
-            bootkernel = None
-
-        if not bootkernel and args.virtType != 'hvm':
-            if args.grub2:
+        setup = None
+        try:
```
Comment (ChrisBr) [inline ec2utils/ec2uploadimg/ec2uploadimg]: I think the indentation is now wrong!
Also, what is the purpose of this change? I don't get it?
That you can define setup inside the loop?

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 5]
```diff
@@ -366,207 +366,208 @@ try:
...
-            except:
-                print('Could not determine helper AMI-ID', file=sys.stderr)
-                sys.exit(1)
-        bootkernel = args.akiID
-        if args.virtType == 'hvm':
-            bootkernel = None
-
-        if not bootkernel and args.virtType != 'hvm':
-            if args.grub2:
+        setup = None
```
Comment (ChrisBr) [inline ec2utils/ec2uploadimg/ec2uploadimg]: This shouldn't be necessary

File: ec2utils/ec2uploadimg/ec2uploadimg (hunk 4 above)
Comment (ChrisBr) [reply ec2utils/ec2uploadimg/ec2uploadimg]: Discussed offline, fine for me

**Comment by mschnitzer** [comment]
@rjschwei open for review again ;-)

**Review by rjschwei — APPROVED** [review]
